### PR TITLE
token-swap js: Prepare package for npm

### PR DESCRIPTION
**Pull request by joncinque**
* Run `pretty` properly
* Add `build:program` like for spl-token
* Add TypeScript to CI
* snake_case -> camelCase variables
* Combine transactions on swap initialization
* Make instruction functions static

### CODE REVIEW

File: token-swap/js/package.json [hunk 1]
```diff
@@ -3,7 +3,7 @@
   "version": "0.0.1",
   "description": "SPL Token Swap JavaScript API",
   "license": "MIT",
-  "author": "Solana Maintainers <maintainers@solana.com>",
+  "author": "Solana Maintainers <maintainers@solana.foundation>",
```
Comment (joncinque) [inline token-swap/js/package.json:6]: Is this correct? I noticed the packages in the monorepo used this address

**Comment by joncinque** [comment]
Please let me know if there are any other things required to get this package out.  I aimed to model this after the token package.

File: token-swap/js/package.json (hunk 1 above)
Comment (mvines) [reply token-swap/js/package.json:6]: yep!  it's correct

**Comment by mvines** [comment]
Do we just want to drop the flow binding?  I'm not opposed

**Review by mvines — APPROVED** [review]
lgtm overall!

**Comment by CriesofCarrots** [comment]
🐫

**Pull request merged**